python/ray/tune/ray_trial_executor.py

```diff
@@ -534,48 +530,20 @@ def on_step_begin(self):
 
     def save(self, trial, storage=Checkpoint.DISK):
         """Saves the trial's state to a checkpoint."""
-        trial._checkpoint.storage = storage
-        trial._checkpoint.last_result = trial.last_result
         if storage == Checkpoint.MEMORY:
-            trial._checkpoint.value = trial.runner.save_to_object.remote()
+            value = trial.runner.save_to_object.remote()
+            checkpoint = Checkpoint(storage, value, trial.last_result)
         else:
-            # Keeps only highest performing checkpoints if enabled
-            if trial.keep_checkpoints_num:
-                try:
-                    last_attr_val = trial.last_result[
-                        trial.checkpoint_score_attr]
-                    if (trial.compare_checkpoints(last_attr_val)
-                            and not math.isnan(last_attr_val)):
-                        trial.best_checkpoint_attr_value = last_attr_val
-                        self._checkpoint_and_erase(trial)
-                except KeyError:
-                    logger.warning(
-                        "Result dict has no key: {}. keep"
-                        "_checkpoints_num flag will not work".format(
-                            trial.checkpoint_score_attr))
-            else:
-                with warn_if_slow("save_to_disk"):
-                    trial._checkpoint.value = ray.get(
-                        trial.runner.save.remote())
-
-        return trial._checkpoint.value
-
-    def _checkpoint_and_erase(self, trial):
-        """Checkpoints the model and erases old checkpoints
-            if needed.
-        Parameters
-        ----------
-            trial : trial to save
-        """
-
-        with warn_if_slow("save_to_disk"):
-            trial._checkpoint.value = ray.get(trial.runner.save.remote())
-
-        if len(trial.history) >= trial.keep_checkpoints_num:
-            ray.get(trial.runner.delete_checkpoint.remote(trial.history[-1]))
-            trial.history.pop()
+            with warn_if_slow("save_checkpoint_to_disk"):
+                value = ray.get(trial.runner.save.remote())
+                checkpoint = Checkpoint(storage, value, trial.last_result)
 
-        trial.history.insert(0, trial._checkpoint.value)
+        with warn_if_slow("commit_checkpoint"):
+            try:
+                trial.commit_checkpoint(checkpoint)
+            except Exception:
+                logger.exception("Error committing checkpoint for Trial %s",
+                                 trial)
 
```

Comment:
we dont need to commit memory checkpoints i think

Reply:
yeah i wasn't, just a func naming issue—changed to be more clear